sound/soc/sof/intel/hda-stream.c

```diff
@@ -201,15 +212,27 @@ int hda_dsp_stream_put(struct snd_sof_dev *sdev, int direction, int stream_tag)
 		if (s->direction == direction &&
 		    s->opened && s->stream_tag == stream_tag) {
 			s->opened = false;
-			spin_unlock_irq(&bus->reg_lock);
-			return 0;
+			found = true;
 		}
+
+		/* check if there are any active capture streams */
+		if (s->direction == SNDRV_PCM_STREAM_CAPTURE && s->opened)
```

Comment:
I think you could do this as
```
} else if (s->direction == SNDRV_PCM_STREAM_CAPTURE) {
```
That would reduce the number of conditions to check and allow you to remove the braces from the list_for_...() loop

Reply:
@lyakh I guess you are missing one point that we could be closing the capture stream and at the same time checking if no capture streams are open. So I cant do an else here. I have to first close the capture stream and then also check at the same time.

Reply:
doing an else if would reduce just the one check for the list iteration but it is less intuitive to follow the logic.

Reply:
@ranj063 to your first comment - I'm not sure what you mean. If the first condition is true and you enter the "if" block, then you set "->opened" to false, so, you don't enter the second "if." So, they are mutually exclusive. Therefore I think doing an "else" actually makes this mutual exclusivity more obvious.